packages/neon_http_client/lib/src/client.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:cookie_store/cookie_store.dart';
+import 'package:http/http.dart' as http;
+import 'package:neon_http_client/src/interceptors/interceptors.dart';
+import 'package:universal_io/io.dart';
+
+/// An exception caused by an error in a [NeonClient].
+sealed class NeonClientException extends http.ClientException {
+  NeonClientException(super.message, [super.uri]);
+}
+
+/// An exception caused by a [HttpInterceptor].
+final class InterceptionFailure extends NeonClientException {
+  /// Creates a new interceptor failure exception.
+  InterceptionFailure(super.message, [super.uri]);
+}
+
+/// A client configuring the clients for all Nextcloud APIs.
+///
+/// To access the APIs of a particular app import the extensions through `package:nextcloud/{id}.dart`.
+final class NeonClient with http.BaseClient {
+  /// Creates a new Neon http client.
+  ///
+  ///
+  /// It is good practice to set the [userAgent] to allow server admins to identify clients.
+  /// A custom HTTP client can be provided through [client].
+  /// Additionally a [cookieStore] can be specified to save cookies across requests.
+  /// Some endpoints require the use of a cookies persistence.
+  NeonClient({
+    http.Client? client,
+    Iterable<HttpInterceptor>? interceptors,
```

Comment:
Do we actually have to expose this? I didn't see any usage of it at the moment and I'd say this client should handle the needs by providing extra parameters like the userAgent or cookieStore which are nicer to use.

Reply:
Ah maybe this is handy for the SVG dimensions rewrite interceptor which doesn't really belong into this package.